### PR TITLE
Bug 105: IdeaPluginPlugin: Bug in collecting jars for ideaSandbox

### PR DESCRIPTION
**Pull request by amityadav1**
Fixing a bug described in Issue: 105.

### CODE REVIEW

File: xtext-idea-gradle-plugin/src/main/java/org/xtext/gradle/idea/IdeaPluginPlugin.xtend [hunk 1]
```diff
@@ -11,8 +11,9 @@ import java.io.File
 
 class IdeaPluginPlugin implements Plugin<Project> {
 	public static val IDEA_ZIP_TASK_NAME = "ideaZip"
-	private static val ARTIFACT_ID = Pattern.compile("(.*?)(-.*)?\\.jar")
-	
+	private static val ARTIFACT_ID = Pattern.compile("(.*?)(-[0-9].*)?\\.jar")
+
+
```
Comment (cdietrich) [inline xtext-idea-gradle-plugin/src/main/java/org/xtext/gradle/idea/IdeaPluginPlugin.xtend]: one newline should be enhough

File: xtext-idea-gradle-plugin/src/main/java/org/xtext/gradle/idea/IdeaPluginPlugin.xtend [hunk 2]
```diff
@@ -38,14 +39,17 @@ class IdeaPluginPlugin implements Plugin<Project> {
 		]
 		project.tasks.getAt(BasePlugin.ASSEMBLE_TASK_NAME).dependsOn(ideaZip)
 	}
-	
+
 	private def hasSameArtifactIdAs(File file1, File file2) {
-		file1.artifactId == file2.artifactId
+		if (file1.artifactId != null && file2.artifactId != null) {
```
Comment (cdietrich) [inline xtext-idea-gradle-plugin/src/main/java/org/xtext/gradle/idea/IdeaPluginPlugin.xtend]: please use `===` respectively `!===` to do null checks

**Comment by cdietrich** [comment]
i wonder if we should move `getArtifactId` and `hasSameArtifactIdAs` to a `IdeaPluginPluginUtils` and add some unit tests. @oehme what do you think

**Comment by amityadav1** [comment]
Ping, Can we please move this review forward.

**Comment by cdietrich** [comment]
from my pov its ok. @oehme i like to have a second opinion from you

**Comment by amityadav1** [comment]
@oehme Could you please review this request ?

**Pull request merged**